index.rst

```diff
@@ -53,10 +53,362 @@
+filters, the number of groups (i.e., the number of instances
+of the full stream flowing in this system) should be kept
+to a minimum.
+See DMTN-028 :cite:`DMTN-028`.
+In this prototype, we recommend here to run five filter
+groups, each acting on a group of 20 filters
+(i.e., 1-20, 21-40, 41-60, 61-80, 81-100).
+Each group of filters can be deployed in its own
+Docker container.
+
+To deploy, e.g., the first group of filters on an
+unfiltered stream called ``full-stream`` on a single node,
+run the following:
+
+.. code-block:: python
+
+    docker run -it --rm \
+               --network=alert_stream_default \
+               alert_stream python bin/filterStream.py full-stream 1 20
+
+Alternatively, as a service in a Docker Swarm, run:
+
+.. code-block:: python
+
+      docker service create \
+              --name filtergroup1 \
+              --network kafkanet \
+              --constraint node.role==worker \
+              -e PYTHONUNBUFFERED=0 \
+              alert_stream python bin/filterStream.py full-stream 1 20
+
+This constrains filter groups to worker nodes, separate from
+Kafka and Zookeeper.
+You can also ensure that each filter group is deployed on its
+own node by taking advantage of constraint ``node.id`` instead.
+
+
+Alerts
+---------------
+
+The alerts used here have realistic content generated by
+the Sims/Commissioning team, but lack object histories
+and stamps.
+(TODO: Add stamps.)
+Alerts are in Avro format with a schema following the
+schema detailed in the DPDD.
+The schema and sample data can be found in the repo
+https://github.com/lsst-dm/sample-avro-alert.
+
+Included in the Docker image are a small number of
+Avro files for testing.
+Each file contains one visit of alerts.
+The alerts were generated without a signal-to-noise
+cut, and therefore each file contains more alerts than
+expected per visit.
+(TODO: Cut on signal-to-noise.)
+However, the alert producer code limits the number of
+alerts sent to Kafka to 10,000 per visit.
+
+
+Alert producers
+---------------
+
+The ``sendAlertStream.py`` script reads Avro files
+from the ``data`` directory and
+produces alerts to Kafka, one visit every 39 seconds.
+The code can use the sample files included in the
+image or files mounted as a volume into a Docker
+container to the internal ``data`` directory.
+In this prototype only one alert producer is used.
+Multiple producers could also be used by scaling up
+the number of Docker containers and modifying the code
+to produce the number of alerts which will yield
+a total of 10,000 across all producers.
+
+To run one alert producer on a single node, mounting
+a local directory of Avro files inside the container, run:
+
+.. code-block:: python
+
+    docker run -it --rm \
+               --name=sender \
+               -v $PWD:/home/alert_stream/data:ro \
+               --network=alert_stream_default \
+               alert_stream python bin/sendAlertStream.py full-stream
+
```

Comment:
This also gives me an error:

```
(base) D-10-155-91-135:data ebellm$ docker run -it --rm \
>            --name=sender \
>            -v $PWD:/home/alert_stream/data:ro \
>            --network=alert_stream_lsst_default \
>            alert_stream python bin/sendAlertStream.py full-stream
usage: sendAlertStream.py [-h] [--encode | --encode-off] topic avrofile
sendAlertStream.py: error: the following arguments are required: avrofile
```

Reply:
Again here need DM-13476.

Reply:
Okay, one more PEBKAC problem: when I run this from the `alert_stream/data` directory it exits silently, sending no alerts.  I'm guessing it's failing to find the packets but I don't know why.

Reply:
My keyboard/chair.  I had wanted to change where the data directory is mounted but never committed it.  Changed and pushed to tickets/DM-13476 so it should work now.  Sorry about that.